src/Adapters/MockAdapter.ts

```diff
@@ -567,90 +572,88 @@ export default class MockAdapter
-                                // if its a TwinToObjectMappingDatasource get the mapping id
-                                if (
-                                    dataSource.type ===
-                                    DatasourceType.ElementTwinToObjectMappingDataSource
-                                ) {
-                                    dataSource.elementIDs.forEach(
-                                        (mappingId) => {
-                                            mappingIds.push(mappingId);
-                                        }
-                                    );
-                                }
+                    const behavior = ViewerConfigUtility.getBehaviorById(
+                        config,
+                        behaviorId
+                    );
+                    if (!behavior) {
+                        continue;
+                    }
+                    const mappingIds: string[] = [];
+                    // cycle through the datasources of behavior
+                    for (const dataSource of behavior.datasources) {
+                        // if its a TwinToObjectMappingDatasource get the mapping id
+                        if (
+                            dataSource.type ===
+                            DatasourceType.ElementTwinToObjectMappingDataSource
+                        ) {
+                            dataSource.elementIDs.forEach((mappingId) => {
+                                mappingIds.push(mappingId);
+                            });
+                        }
 
-                                // TODO get FilteredTwinDatasources
-                            }
+                        // TODO get FilteredTwinDatasources
+                    }
 
-                            // cycle through mapping ids to get twins for behavior and scene
-                            for (const id of mappingIds) {
-                                const twins = {};
-                                const element: ITwinToObjectMapping = scene.elements.find(
-                                    (mapping) =>
-                                        mapping.type ===
-                                            ElementType.TwinToObjectMapping &&
-                                        mapping.id === id
-                                ) as ITwinToObjectMapping;
-
-                                if (element) {
-                                    // get primary twin
-                                    twins[
-                                        PRIMARY_TWIN_NAME
-                                    ] = this.mockTwins.find(
-                                        (t) => t.$dtId === element.primaryTwinID
+                    // cycle through mapping ids to get twins for behavior and scene
+                    for (const id of mappingIds) {
+                        const twins = {};
+                        const element: ITwinToObjectMapping = scene.elements.find(
```

Comment:
if there is no elements in scene it is always an empty array, right? Otherwise:
```suggestion
                        const element: ITwinToObjectMapping = scene.elements?.find(
```